### ISSUE TITLE
Support side-loaded ComicInfo.xml files

### ISSUE THREAD

**Issue by Maltavius** [comixed/comixed#1559]
**Is your feature request related to a problem? Please describe.**
I have a huge collection of Comics that I currently manage with ComicRack, I save all the metadata in XML-files that are the same name as the comic-file (only the extension is different). Since I don't want to modify the CBR/CBZ files since then their checksums become different.

**Describe the solution you'd like**
I'd like for ComiXed to look for files named the same as a CBR/CBZ file with a ComicInfo.xml format and read that instead of looking for the file inside the archive. 

**Describe alternatives you've considered**
There is no alternative, I won't modify my files.

**Additional context**
Emby/Jellyfin/Kodi and other media managers have worked like this for year.

**Comment by mcpierce** [comment]
@Maltavius I'm wondering about the those files are ultimately being used? Does Emby, Jellyfin, and Kodi use those files to generate the navigation feeds for OPDS readers, and/or does something external to them ingest the data from those files? 

If the former, then CX already does this, since it generates the feeds from the database and not the physical files and not create a separate metadata file for each comic.

If the latter, then we can definitely add a feature to optionally generate external files for the metadata when the library is consolidated or the metadata is updated.

And since you don't want to modify your comic files, another separate configurable option to explicitly create ComicInfo.xml in the comic files. That way you could turn off that feature and not have it show you buttons to update the comic files when the metadata changes.

**Comment by Maltavius** [comment]
> @Maltavius I'm wondering about the those files are ultimately being used? Does Emby, Jellyfin, and Kodi use those files to generate the navigation feeds for OPDS readers, and/or does something external to them ingest the data from those files?
> 
> If the former, then CX already does this, since it generates the feeds from the database and not the physical files and not create a separate metadata file for each comic.
> 
> If the latter, then we can definitely add a feature to optionally generate external files for the metadata when the library is consolidated or the metadata is updated.
> 

The files in Emby/Plex/Jellyfin etc. are for Movies and containt the description of the movie, different ratings and meta-data about the movie. Each Media Manager then reads in that to their database.
If there are no .nfo-files the Media Manager will scrape meta-data from the file themselves and fetch information from thetvdb.com/imdb.com or similar.
This is especially useful when moving between different Media Managers, I already know my files have the correct meta-data.

There are several locations with information about NFO-files.
https://jellyfin.org/docs/general/server/metadata/nfo/
https://kodi.wiki/view/NFO_files

I'd want the same for Importing Comics into ComiXed, I already know my metadata is correct and for those I have external XML-files. For those I don't need ComiXed to go out and fetch all the data remotely, only missing data that ComiXed needs.

> And since you don't want to modify your comic files, another separate configurable option to explicitly create ComicInfo.xml in the comic files. That way you could turn off that feature and not have it show you buttons to update the comic files when the metadata changes.

Well yes, that's a must. Are you saying ComiXed automatically creates a ComixInfo.xml in my files when I import them? Without asking?

**Comment by mcpierce** [comment]
@Maltavius No, CX only updates the ComicInfo.xml file when you tell it to do so: it does not change files without being explicitly told to do so by clicking the "Update File" button either on the comic details page or else on a list view when comics are selected.

As for the external files, would the file's name be something like:
1. if the comic is named "Batman v2011 #0 (Nov 2012).cbz",
1. then the external metadata file would be named "Batman v2011 #0 (Nov 2012).nfo".

And the content would be the usual ComicInfo.xml tagset. Is that correct?

If so, I'd be happy to add this feature for our June release (v1.5). I'll need to split it into a couple of tickets (one to generate the files on scraping/consolidation, one for ingesting the file if present when importing, and one for adding the configuration options to enable this and to disable updating comic files directly).

**Comment by Maltavius** [comment]
@mcpierce That's great!

The files that are exported from the ComicRack Exporter names them with the extension .xml.

For example:
```
Age of Conan - Belit, Queen of the Black Coast 04 (of 05) (2019).cbr
Age of Conan - Belit, Queen of the Black Coast 04 (of 05) (2019).xml
```
But I guess handling both .xml and .nfo wouldn't be a negative.

The contents starts with this (which I guess is the ordinary ComixInfo.xml format)

```
<?xml version="1.0"?>
<ComicInfo xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <Title>Chapter 4: The Oath</Title>
  <Series>Age of Conan: Bêlit, Queen of the Black Coast</Series>
  <Number>4</Number>
  <Volume>2019</Volume>
  <Summary>.....
```

**Comment by mcpierce** [comment]
@Maltavius Yep, that looks like a run-of-the-mill comicinfo.xml file's contents.

I'll create the issues to track those features, and will plan to have them in v1.5 in June. Thanks for the suggestions!

**Comment by mcpierce** [comment]
@Maltavius The creating and moving of the external metadata files is complete and will be in the next release in June!

**Comment by Maltavius** [comment]
Thanks
And if I already have files those will just be read into the database?

Den lör 29 apr. 2023 23:52Darryl L. Pierce ***@***.***> skrev:

**Comment by mcpierce** [comment]
> Thanks
> And if I already have files those will just be read into the database?

Ultimately yes. I have another feature to write that will load the metadata from a file if it exists with the same base filename as the comic. That should do what you're wanting. It'll only work when importing the comics though since, once imported, the source of truth for metadata is the database.

**Comment by Maltavius** [comment]
> > Thanks
> > And if I already have files those will just be read into the database?
> 
> Ultimately yes. I have another feature to write that will load the metadata from a file if it exists with the same base filename as the comic. That should do what you're wanting. It'll only work when importing the comics though since, once imported, the source of truth for metadata is the database.

Which Issue should I follow to know when this is implemented?

**Comment by mcpierce** [comment]
> Which Issue should I follow to know when this is implemented?

That was #1601, which will be in the next release at the end of the month.

**Comment by Maltavius** [comment]
Ok, thanks for the clarification.

**Comment by mcpierce** [comment]
> Ok, thanks for the clarification.

Sorry I misspoke before. That feature was included in v1.5 back in June.

What is the filename pattern used for external meta-files? The code is looking for files that have the same basename as the comic, but with a ".meta" file extension.

**Comment by Maltavius** [comment]
> > Ok, thanks for the clarification.
> 
> Sorry I misspoke before. That feature was included in v1.5 back in June.
> 
> What is the filename pattern used for external meta-files? The code is looking for files that have the same basename as the comic, but with a ".meta" file extension.

Yeah that's it then... the Feature request was for it to work with the XML extension as in this [comment](https://github.com/comixed/comixed/issues/1559#issuecomment-1478644711)

I have no .meta files and no way of generating 5000 .meta files for my Comics.

**Comment by mcpierce** [comment]
> Yeah that's it then... the Feature request was for it to work with the XML extension as in this [comment](https://github.com/comixed/comixed/issues/1559#issuecomment-1478644711)
> 
> I have no .meta files and no way of generating 5000 .meta files for my Comics.

Ah! I'll update the app for the next release to use .xml instead of .meta for the extension.
